web/app/habit/history/components/HistoryContent.tsx

```diff
@@ -24,18 +27,40 @@ function HistoryContent() {
     [challenges],
   );
 
+  const sortedChallenges = sortChallenges(finishedChallenges, sortBy);
+
   return (
     <div className="flex h-screen w-full flex-col items-center justify-start">
-      {/* if no challenges, show new user component */}
-      <div className="flex w-full items-center justify-center">
-        <p className="my-4 font-londrina text-xl font-bold">
-          {' '}
-          {finishedChallenges.length > 0 ? 'Challenge History' : 'No Challenge History'}{' '}
+      <div className="relative flex w-full items-center justify-between px-4">
+        <div className="w-1/3" /> {/* Spacer */}
+        <p className="absolute left-1/2 my-4 -translate-x-1/2 transform font-londrina text-xl font-bold">
+          {sortedChallenges.length > 0 ? 'Challenge History' : 'No Challenge History'}
         </p>
+        <div className="flex w-1/3 justify-end">
+          {' '}
+          {/* Right-aligned container */}
+          <Dropdown>
+            <DropdownTrigger>
+              <Button variant="light" size="sm">
+                <FaSort className="mr-2" />
+                Sort
+              </Button>
+            </DropdownTrigger>
+            <DropdownMenu
+              aria-label="Sort challenges"
+              onAction={(key) => setSortBy(key as SortOption)}
+            >
+              <DropdownItem key="status">Status</DropdownItem>
+              <DropdownItem key="stakes">Stakes</DropdownItem>
+              <DropdownItem key="startTimestamp">Start Date</DropdownItem>
+              <DropdownItem key="endTimestamp">End Date</DropdownItem>
```

Comment:
I think pass in a array like the example in dynamic style would be cleaner

https://nextui.org/docs/components/dropdown#dynamic-items